src/core/documentation/documentation.ts

```diff
@@ -0,0 +1,158 @@
+import { Model } from "../model/model";
+import { Element } from "../model/element";
+import * as assert from "assert";
+import { Section } from "./section";
+import { Decision } from "./decision";
+import { SoftwareSystem } from "../model/softwareSystem";
+import { DecisionStatus } from "./decisionStatus";
+import { Format } from "./format";
+
+export class Documentation {
+    private model: Model;
+    public sections: Set<Section> = new Set();
+    public decisions: Set<Decision> = new Set();
+
+    constructor(model: Model) {
+        this.model = model;
+    }
+
+    public addSection(
+        element: Element,
+        title: string,
+        format: any,
+        content: string
+    ) {
+        if (element && !this.model.containsElement(element)) {
+            throw new Error(
+                `The element named ${element.name} does not exist in the model associated with this documentation.`
+            );
+        }
+
+        assert(title, "A title must be specified.");
+        assert(content, "Content must be specified.");
+        assert(format, "A format must be specified.");
+        this.checkSectionIsUnqiue(title, element);
+
+        const section = new Section();
+        section.fromDto({
+            element,
+            title,
+            order: this.sections.size + 1,
+            format,
+            content
+        });
+        this.sections.add(section);
+        return section;
+    }
+
+    public addDecision(
+        element: SoftwareSystem,
```

Comment:
Same here: the softwareSystem should be optional (and better renamed): `softwareSystem: SoftwareSystem | undefined`.

Also: the parameter is not used in the code below, which I believe it should, right?